lib/chart/gauge/settings.ex

```diff
@@ -35,18 +35,18 @@ defmodule Chart.Gauge.Settings do
             key_guard(config, :major_ticks_count, 7, &Validators.validate_major_ticks_count/1),
           gap: key_guard(config, :major_ticks_gap, 0, &Validators.validate_number/1),
           length:
-            key_guard(config, :major_ticks_length, 7, &Validators.validate_positive_number/1)
+            key_guard(config, :major_ticks_length, 5, &Validators.validate_positive_number/1)
         }
-        |> set_translate(settings.gauge_center)
+        |> set_translate(settings.gauge_center, settings.gauge_radius)
```

Comment:
Pass only `settings` struct. The function will be simple.

Reply:
I think it depends on point of view. Calling `set_translate(settings)` function will be simple, but her definition already won't + necessary `alias ...Settings` in this submodule. Compare
` defp set_translate(major_ticks, {cx, cy}, {rx, _ry}) do`
with
` defp set_translate(major_ticks, %Settings{gauge_center: {cx, cy}, gauge_radius: {rx, ry}},) do`

Reply:
Not necessarily. Add for free _design by contract_ and follow a _crate pattern._